pkg/client/request_test.go

```diff
@@ -748,6 +742,64 @@ func TestDoRequestNewWay(t *testing.T) {
 	}
 }
 
+func TestCheckRetryClosesBody(t *testing.T) {
```

Comment:
Maybe I'm dense, but how does this actually test that the client closes the body properly?